lib/license_scout/dependency_manager/habitat.rb

```diff
@@ -0,0 +1,126 @@
+#
+# Copyright:: Copyright 2018, Chef Software Inc.
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require "license_scout/dependency_manager/base"
+
+require "open-uri"
+require "mixlib/shellout"
+
+module LicenseScout
+  module DependencyManager
+    class Habitat < Base
+
+      def name
+        "habitat"
+      end
+
+      def type
+        "habitat"
+      end
+
+      def signature
+        File.exist?(habitat_plan_sh_path) ? "habitat/plan.sh file" : "plan.sh file"
+      end
+
+      def install_command
+        ""
+      end
+
+      def detected?
+        File.exist?(plan_sh_path) || File.exist?(habitat_plan_sh_path)
+      end
+
+      def dependencies
+        tdeps = Set.new(pkg_deps)
+
+        pkg_deps.each do |pkg_dep|
+          pkg_info(pkg_dep)["tdeps"].each { |dep| tdeps << to_ident(dep) }
+        end
+
+        tdeps.sort.map do |tdep|
+          o, n, v, r = tdep.split("/")
+          dep_name = "#{o}/#{n}"
+          dep_version = "#{v}-#{r}"
+
+          dependency = new_dependency(dep_name, dep_version, nil)
+
+          license_from_manifest(pkg_info(tdep)["manifest"]).each do |spdx|
+            dependency.add_license(spdx, "https://bldr.habitat.sh/v1/depot/channels/#{o}/stable/pkgs/#{n}/#{v}/#{r}")
+          end
+
+          dependency
+        end.compact
+      end
+
+      private
+
+      def license_from_manifest(manifest_content)
+        /^*\s+__License__:\s+(.+)$/.match(manifest_content)[1].strip.split("\s")
+      end
+
+      def pkg_deps
+        @pkg_deps ||= begin
+          plan_path = File.exist?(plan_sh_path) ? plan_sh_path : habitat_plan_sh_path
+
+          c = Mixlib::ShellOut.new("source #{plan_path}; echo $pkg_deps", LicenseScout::Config.environment)
```

Comment:
What version of bash did you try this with? I had to do something like `echo ${pkg_deps[*]}`

Reply:
That's a good question - I just ran it locally. It's probably worth me circling back to that.